### PR TITLE
fix: QoL improvements to make the SDK more stable

### LINKED ISSUE #1036: investigate ways to clean sdk
with a myriad of integrations now, we should be in a position to identify patterns and abstractions between integrations

lets see how we can simplify and clean up the sdk to make it more maintainable and extensible for the future :)

### PR DESCRIPTION
**Pull request by dot-agi**
## 📥 Pull Request

**📘 Description**
Enhanced AgentOps SDK reliability, consistency, and developer experience through comprehensive improvements across core functionality, error handling, and dependency management.

### 🔧 Core Infrastructure Improvements
- **Enhanced Error Handling**: Improved `TracingCore._flush_span_processors()` with comprehensive exception handling for `AttributeError`, `RuntimeError`, and unexpected errors, enhanced logging levels, and graceful degradation to ensure span processor failures don't break application flow
- **Code Consolidation**: Added `format_trace_id()` utility function to eliminate duplicate trace ID formatting logic across the codebase, improving maintainability and consistency
- **Resource Management**: Removed deprecated `LiveSpanProcessor` class to simplify the processor architecture and reduce maintenance overhead

### 🎯 API Consistency & Backward Compatibility
- **Decorator Improvements**: Enhanced `@trace` decorator with better error handling, proper cleanup in finally blocks, and improved async/sync function support
- **Deprecation Management**: Added proper deprecation warnings for `@session` decorator while maintaining backward compatibility
- **Session Lifecycle**: Improved trace context management with better cleanup and state tracking
- **Python Compatibility**: Added `typing_extensions>=4.0.0` for Python <3.11 compatibility, ensuring `Required` and `NotRequired` types work across all supported Python versions

### 🛡️ Reliability & Error Resilience
- **Graceful Degradation**: Enhanced error handling ensures that telemetry failures don't break user applications
- **Resource Leak Prevention**: Improved cleanup mechanisms in trace management to prevent resource leaks
- **Logging Improvements**: Better structured logging with appropriate levels (debug, warning, error) for different failure scenarios

**🧪 Testing**
Comprehensive test coverage added to validate all improvements:

- **50+ new unit tests** across 2 test modules covering edge cases, error conditions, and integration scenarios
- **Error Resilience Testing**: Exception handling for AttributeError, RuntimeError, and unexpected errors during span processor operations (`test_core_error_handling.py`)
- **API Consistency Testing**: Return value behavior validation, parameter passing, backward compatibility patterns, and exception propagation (`test_init_api_consistency.py`)
- **Integration Testing**: Real-world usage patterns, shutdown behavior during active span processing, and graceful degradation scenarios
- **Backward Compatibility**: Ensures existing user code continues to work without modifications

**🔄 Migration Impact**
- **Zero Breaking Changes**: All changes maintain backward compatibility
- **Automatic Benefits**: Users get improved reliability and error handling without code changes
- **Optional Upgrades**: New dependency versions are available in `dev-llm` group for development environments
- **Deprecation Path**: Clear migration path from `@session` to `@trace` decorator with helpful warnings

These changes address critical production issues including resource management, error propagation during shutdown scenarios, and dependency version gaps while maintaining full backward compatibility and improving the overall developer experience.

### CODE REVIEW

File: agentops/sdk/types.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from typing import Annotated, Optional, TypedDict
+from typing import Annotated, Optional, Required, TypedDict
```
[the remark this answers is not in the document]
Comment (dot-agi) [reply agentops/sdk/types.py]: fixed

File: agentops/sdk/decorators/__init__.py [hunk 2]
```diff
@@ -16,15 +15,21 @@
...
 tool = create_entity_decorator(SpanKind.TOOL)
-operation = task
 
-# For backward compatibility: @session decorator calls @trace decorator
-@functools.wraps(trace)
+
+# For backward compatibility: @session decorator calls @trace decorator with deprecation warning
 def session(*args, **kwargs):
     """@deprecated Use @agentops.trace instead. Wraps the @trace decorator for backward compatibility."""
+    import warnings
```
Comment (Dwij1704) [inline agentops/sdk/decorators/__init__.py]: Redundant warning

File: pyproject.toml
Comment (Dwij1704) [inline pyproject.toml:1]: If updating pyproject.toml, might as well update dev dependencies to include all instrumented packages

File: agentops/sdk/processors.py
Comment (Dwij1704) [inline agentops/sdk/processors.py:1]: LiveSpanProcessor is redundant, lets clean it up if not needed

File: tests/unit/sdk/test_live_span_processor_lifecycle.py
Comment (Dwij1704) [inline tests/unit/sdk/test_live_span_processor_lifecycle.py:1]: No need for LiveSpanProcessor tests

**Review by Dwij1704 — CHANGES_REQUESTED** [review]

File: agentops/sdk/decorators/__init__.py (hunk 2 above)
Comment (dot-agi) [reply agentops/sdk/decorators/__init__.py]: Refactored to use `logger`.

File: agentops/sdk/processors.py
Comment (dot-agi) [reply agentops/sdk/processors.py:1]: Done.

File: tests/unit/sdk/test_live_span_processor_lifecycle.py
Comment (dot-agi) [reply tests/unit/sdk/test_live_span_processor_lifecycle.py:1]: Done.

File: pyproject.toml
Comment (dot-agi) [reply pyproject.toml:1]: Created `dev-llm` for separation of concerns since -
* `dev` is meant to be used for testing of the SDK functionality
* `test` is meant for CI which means unit and integration tests, where the LLM providers and agentic libraries must be present
* This would be a temporary group which will be removed when #1017 is completed.

File: agentops/sdk/decorators/factory.py [hunk 3]
```diff
@@ -21,6 +21,114 @@
...
+def _handle_session_trace_sync(
+    operation_name: str, tags: Optional[Union[list, dict]], wrapped_func: Callable, args: tuple, kwargs: Dict[str, Any]
+) -> Any:
+    """Helper function to handle SESSION trace lifecycle for sync functions with proper cleanup"""
+    trace_context: Optional[TraceContext] = None
+    trace_ended = False
+
+    try:
+        # Start trace
+        trace_context = TracingCore.get_instance().start_trace(trace_name=operation_name, tags=tags)
```
Comment (braelyn-ai) [inline agentops/sdk/decorators/factory.py]: use the new `tracer` variable pls #1030

Comment (dot-agi) [reply agentops/sdk/decorators/factory.py]: Done.

**Review by braelyn-ai — APPROVED** [review]

**Comment by dot-agi** [comment]
Closing this because it's scraping the bottom of the barrel. Too many conflicts to invest time into and a better quality PR is needed.
